splitgraph/core/repository.py

```diff
@@ -1078,6 +1133,25 @@ def diff(
         # TODO we can aggregate chunks in a similar way that LQ does it.
         return slow_diff(self, table_name, _hash(image_1), _hash(image_2), aggregate)
 
+    def is_overlay_view(self, table_name: str) -> bool:
+        """
+        Check whether the provided table is actually the write overlay mechanism, merging lower and upper/staging table.
+        """
+        from splitgraph.hooks.data_source.base import (
+            WRITE_LOWER_PREFIX,
+            WRITE_UPPER_PREFIX,
+        )
+
+        is_view = self.object_engine.get_table_type(self.to_schema(), table_name) == "VIEW"
+        lower_table_exists = self.object_engine.table_exists(
+            self.to_schema(), WRITE_LOWER_PREFIX + table_name
+        )
+        upper_table_exists = self.object_engine.table_exists(
+            self.to_schema(), WRITE_UPPER_PREFIX + table_name
+        )
+
+        return is_view and lower_table_exists and upper_table_exists
```

Comment:
You could golf this and do 

```python
return self.object_engine.get_table_type(self.to_schema(), table_name) == "VIEW" \
    and self.object_engine.table_exists(...) \
    and ...
```

(which will have the performance effect of short-circuiting if the table isn't a view, running one query instead of three)